server/api.go

```diff
@@ -669,14 +693,34 @@ func (a *apiServer) subscribe(ctx context.Context, partition *partition,
 				}
 				return
 			}
+			msgValue := m.Value()
+
 			headers := m.Headers()
+
+			// Data decryption
```

Comment:
This might not need addressed in this initial PR, but I do wonder if this is the right place for decryption to happen or if it shouldn't be handled at a different level, e.g. some kind of encryption wrapper around the commit log. Just something to think about...

Reply:
Agreed. At first I hesitated  about the best place to do encryption/decryption.

I was thinking I can do it at a very low level. i.e: right before writing log segments to files. That would basically sets a pattern for any other logs to follow. However, do such thing would requires a much bigger changes in the PR, while not yet fully understand the low level log segments read/write, I would say this current approach is "safer".

Leaving the encryption/decryption and partition level, though not the best way to do, achieves a better isolation, as at first user can deactivate (by default) encryption and we are sure 100% that this feature cannot affect partitions with encryption deactivated.

However, I still think that we do need a better place to put the encryption/decryption.

Reply:
I think it is ok for now. I believe we can safely refactor this in the future as the API and disk contents will remain the same.